javascript/modules/client/src/client-addressList.ts

```diff
@@ -1,66 +1,19 @@
-import {
-  bytesToHex,
-  GraphQLError,
-  InvalidAddressError,
-  InvalidAddressOrEnsError,
-  InvalidProposalIdError,
-  NoProviderError,
-  Random,
-} from "@aragon/sdk-common";
 import { ContextPlugin } from "./context-plugin";
-import { ClientCore } from "./internal/core";
-import {
-  decodeAddMemebersAction,
-  decodeRemoveMemebersAction,
-  decodeUpdatePluginSettingsAction,
-  encodeAddMembersAction,
-  encodeAddressListActionInit,
-  encodeRemoveMembersAction,
-  encodeUpdatePluginSettingsAction,
-  getFunctionFragment,
-} from "./internal/encoding/plugins";
 import {
-  DaoAction,
-  GasFeeEstimation,
-  IInterfaceParams,
-  IPluginInstallItem,
-  SortDirection,
-} from "./internal/interfaces/common";
-import {
-  AddressListProposal,
-  AddressListProposalListItem,
-  ExecuteProposalStep,
-  ExecuteProposalStepValue,
   IAddressListPluginInstall,
-  ICanVoteParams,
   IClientAddressList,
-  ICreateProposalParams,
-  IExecuteProposalParams,
-  IPluginSettings,
-  IProposalQueryParams,
-  IVoteProposalParams,
-  ProposalCreationSteps,
-  ProposalCreationStepValue,
-  ProposalMetadata,
-  ProposalSortBy,
-  SubgraphAddressListProposal,
-  SubgraphAddressListProposalListItem,
-  VoteProposalStep,
-  VoteProposalStepValue,
+  IClientAddressListDecoding,
+  IClientAddressListEncoding,
+  IClientAddressListEstimation,
+  IClientAddressListMethods,
 } from "./internal/interfaces/plugins";
-import { delay } from "./internal/temp-mock";
-import {
-  computeProposalStatusFilter,
-  isProposalId,
-  toAddressListProposal,
-  toAddressListProposalListItem,
-} from "./internal/utils/plugins";
-import { isAddress } from "@ethersproject/address";
-import {
-  QueryAddressListProposal,
-  QueryAddressListProposals,
-} from "./internal/graphql-queries/proposal";
-import { QueryAddressListPluginSettings } from "./internal/graphql-queries/settings";
+import { IClientAddressListMethodsModule } from "./modules-addressList/methods";
+import { IClientAddressListEncodingModule } from "./modules-addressList/encoding";
+import { IClientAddressListDecodingModule } from "./modules-addressList/decoding";
+import { IClientAddressListEstimationModule } from "./modules-addressList/estimation";
```

Comment:
I would structure files and code differently. I would have `src/internal/<name>/<module>.ts` or similar

Reply:
So move the client ones to `internal/modules` where the `ipfs`, `web3`, and `graphql` modules live and the other ones to `internal/modules-addressList`?

Reply:
no, I was meaning something more like:
- `src/internal/client/index.ts`
- `src/internal/client/methods.ts`
- `src/internal/client/encoding.ts`
- ...
- `src/internal/client-erc20/index.ts`
- `src/internal/client-erc20/methods.ts`
- `src/internal/client-erc20/encoding.ts`
- ...